parquet-hadoop/src/main/java/org/apache/parquet/hadoop/metadata/BlockMetaData.java

```diff
@@ -46,10 +46,17 @@ public BlockMetaData() {
 
   // Reader side (get parameters from RowGroup structure)
   public BlockMetaData(long fileOffset, long totalCompressedSize) {
-    this.startingPosition = fileOffset;
```

Comment:
In the current mr code, this constructor is called from one place only, on the reader path,
ParquetMetadataConverter.fromParquetMetadata
and never with 0 values.

Are you adding these checks for another mr code version (eg with changes for Presto?). If yes, please let me know, and please send a commit for that. I think its best to close the current pr#5 (still shows changes in 3 files..), and open a new pr for the BlockMetaData constructor checks.

Reply:
No this is not presto. We hit this issue in Hive. The problem is that if fileOffset and totalCompressedSize are absent in the parquet file, the current code still think startingPositionSet and totalCompressedSizeSet  are set, and use these two variables. It causes the split is 0 which is wrong. 

This is a critical fix. It causes our Hive pipelines failed. With this fix, the reading succeeded and pipeline pass.

Reply:
sounds good. please open a new pull request with this fix.

Reply:
With this pull request open, how do I do that? I searched around but seems GitHub won't allow me to do that.

Reply:
I'll close this one now, please try again. if doesnt work, please either create a new branch for new pr, or sync this branch to current upstream, and add this fix.

Reply:
Just created another one: https://github.com/ggershinsky/parquet-mr/pull/7